KY-026_FlameSensor/KY-026_FlameSensor.ino

```diff
@@ -1,45 +1,144 @@
+ * albeit with different variable names.
+ *
+ * Each sensor is equipped with:
+ * two output pins: digital (DO) and analog (AO).
+ * Two on-board LEDs: Power (LED1) & Threshold (LED2)
 
-    This is basically an infrared sensor (a flame is an IR source).
-
-    While this is probably not the best solution for detecting a fire, it could be used to detect a pilot light or an overheating component.
-
-    Just like the Reed Switch and the Linear Hall Switch, this switch has both an DIGITAL and an ANALOG output
-
-    The digital value is HIGH if IR is detected and LOW if not.
-
-    The analog output is 1023 in total darkness and somewhat lower when other lights are present. The analog value DECREASES with the intensity if the IR source.
-
-    You can also trigger this sensor with the same IR remote you used in the IR remote sensor project.
+ * When the intensity of the signal is GREATER than the set threshold:
+ * DO == 1, AO < ~500, LED2 is ON
+ 
+ * When the intensity of the signal is LESS than the set threshold:
+ * DO == 0, AO > ~500, LED2 is OFF
+ 
+ * The analog pin output (AO) is a range from 0 to 1023
+ * The digital pin output (DO) is either 0 or 1
+ * These are sensors that detects a physical signal: noise, heat, magnetic field... etc
+ * Each sensor has a limited range of detection. 
+ *
+ * ** THE STRONGER THE SIGNAL, THE LOWER THE Analog Value -Closer to 0 **
+ * ** THE WEAKER THE SIGNAL, THE HIGHER THE Analog Value -Closer to 1024 **
+ * ** (AO) LESS THAN ~512 triggers (DO) to 1
+ * ** (AO) GREATER than ~512 triggers (DO) to 0
+ *
+ * These sensors are not calibrated in any meaningful  way.
+ * They are detectors as opposed to calibrated sensors; their outputs are not meant to
+ * be converted to functional units (dB, degrees, lumens, Teslas... etc)
+ *
+ * This project uses:
+ * One Digital pin to OUTPUT (HIGH or LOW)
+ * One Analog pin to OUTPUT (Magnetic field strength & polarity)
+ * The onboard LED
+ * On the Hero (Arduino Uno compatible) we *could* use: D0-D13, A0-A5.
+ * Skip: A0-A5 (can use for any ANALOG pin),
+ *       D0/D1 (used by USB),
+ *       D2/D3 (save for interrupts),
+ *       D13 (used by LED_BUILTIN and SPI Clock),
+ *       D5, D6, D9, D10 and D11 (save for PWM)
+ *       D11 (SPI MOSI)
+ *       D12 (SPI MISO)
+ * Recommended for fewest conflicts:
+ *    Digital pin: D4, D7 or D8
+ *    Analog pin:  A0-A5
  */
 
-int led = 13;        // define the LED pin
-int digitalPin = 2;  // KY-026 digital interface
-int analogPin = A0;  // KY-026 analog interface
-int digitalVal;      // digital readings
-int analogVal;       //analog readings
+const uint8_t KY_026_DIGITAL_OUT = 8;  // digital sensor
+const uint8_t KY_026_ANALOG_OUT = A0;  // analog sensor
 
 void setup() {
-  pinMode(led, OUTPUT);
-  pinMode(digitalPin, INPUT);
-  //pinMode(analogPin, OUTPUT);
   Serial.begin(9600);
+  pinMode(LED_BUILTIN, OUTPUT);  // define LED_BUILTIN as output interface
```

Comment:
Even though the default mode is input, for these lessons it may be good to explicitely set the pinMode() of all used pins.